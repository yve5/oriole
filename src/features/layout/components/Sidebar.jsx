import React from 'react';
import { Link } from 'react-router-dom';

import Logo from '../../header/resources/logo.webp';
import { BoxArrowRight } from '../../icons';

const Sidebar = () => (
  <div
    style={{ width: '280px' }}
    className="d-flexaze flex-columnaze flex-shrink-0aze p-3 bg-darkaze text-whiteaze border-end"
  >
    <ul className="nav nav-pills flex-column mb-auto">
      <li className="nav-item">
        <Link to="/" className="nav-link active">
          <BoxArrowRight />
          Paging
        </Link>
      </li>

      <li className="nav-item">
        <Link to="/buttons" className="nav-link">
          <BoxArrowRight />
          Buttons
        </Link>
      </li>
    </ul>

    <hr />

    <ul className="nav nav-pills flex-column mb-auto">
      <li className="nav-item">
        <button type="button" className="nav-link">
          <BoxArrowRight />
          Expand
        </button>
      </li>
    </ul>
  </div>
);

export default Sidebar;
