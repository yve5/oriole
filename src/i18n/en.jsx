const I18N_DICTIONARY = {
  LANG: 'en',
  LANGUAGE: 'English',
  DATE_FORMAT: 'MM/dd/yyyy',
  LOCALE: 'enUS',
  'About us': 'About us',
  'An error has been detected. Please try again.':
    'An error has been detected. Please try again.',
  Buttons: 'Buttons',
  Changelog: 'Changelog',
  Collapse: 'Collapse',
  Expand: 'Expand',
  Internationalization: 'Internationalization',
  Paging: 'Paging',
  'The request has been sent successfully.':
    'The request has been sent successfully.',
};

export default I18N_DICTIONARY;
