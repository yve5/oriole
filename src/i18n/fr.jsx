const I18N_DICTIONARY = {
  LANG: 'fr',
  LANGUAGE: 'Français',
  DATE_FORMAT: 'dd/MM/yyyy',
  LOCALE: 'fr',
  'About us': 'À propos de nous',
  'An error has been detected. Please try again.':
    'Une erreur a été détectée. Merci de recommencer.',
  Buttons: 'Boutons',
  Changelog: 'Journal des modifications',
  'Changes have been saved.': 'Les changements ont été sauvegardés.',
  'Choose a file': 'Choisir un fichier',
  Collapse: 'Réduire',
  Columns: 'Colonnes',
  'Creation date': 'Date de création',
  'Descending order': 'Ordre décroissant',
  Disabled: 'Désactivé',
  Enabled: 'Activé',
  Expand: 'Développer',
  First: 'Premier',
  'Hello World': 'Bonjour le monde',
  'Hide columns': 'Cacher les colonnes',
  Internationalization: 'Internationalisation',
  Paging: 'Pagination',
  'The request has been sent successfully.':
    'La demande a été envoyée avec succès.',
};

export default I18N_DICTIONARY;
