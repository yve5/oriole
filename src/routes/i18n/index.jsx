import I18nPage from './components/I18n';

export default I18nPage;
