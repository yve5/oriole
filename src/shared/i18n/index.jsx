import LangSwitch from './components/LangSwitch';
import I18n from './components/I18n';
import i18nReducer from './reducer';

export { I18n, LangSwitch, i18nReducer };
