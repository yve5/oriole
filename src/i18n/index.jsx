import fr from './fr';
import en from './en';
import { tsl } from '../shared/i18n/utils/Translate';

const i18n = {
  tsl: (term, args) => tsl(fr, term, args),
  dictionaries: [fr, en],
  lang: fr.LANG,
};

export default i18n;
