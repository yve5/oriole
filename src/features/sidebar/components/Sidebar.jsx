import React, { useState } from 'react';
import { Link } from 'react-router-dom';
import { clsx } from 'clsx';

import * as AllIcons from '../../icons';
import { I18n } from '../../../shared/i18n';
import { SIDEBAR_LINKS } from '../resources/constants';

const Sidebar = () => {
  const [expanded, setExpanded] = useState(false);

  const size = expanded ? 20 : 24;
  const CollapseIcon = AllIcons[expanded ? 'ChevronLeft' : 'ChevronRight'];

  return (
    <div className={clsx('d-none', 'd-lg-flex', 'flex-column', 'border-end')}>
      <ul
        className={clsx(
          'flex-grow-1',
          'nav',
          'nav-pills',
          'flex-column',
          'mb-auto',
          expanded ? 'p-2' : 'text-center'
        )}
      >
        {SIDEBAR_LINKS.map(({ active, icon, route, title }) => {
          const Icon = AllIcons[icon];

          return (
            <li key={title} className={clsx('nav-item')}>
              <Link
                to={route}
                className={clsx(
                  'nav-link',
                  !expanded && ['py-3', 'border-bottom', 'rounded-0'],
                  { active }
                )}
              >
                <I18n title>
                  <Icon
                    width={size}
                    height={size}
                    title={title}
                    className={clsx({ 'me-2': expanded })}
                  />
                </I18n>
                {expanded && <I18n>{title}</I18n>}
              </Link>
            </li>
          );
        })}
      </ul>

      <ul
        className={clsx(
          'nav',
          'nav-pills',
          'flex-column',
          'mb-auto',
          expanded ? 'p-3' : 'text-center'
        )}
      >
        <li className="nav-item">
          <button
            type="button"
            onClick={() => setExpanded(!expanded)}
            className={clsx('nav-link', !expanded && ['py-3', 'rounded-0'])}
          >
            <CollapseIcon
              width={size}
              height={size}
              className={clsx({ 'me-2': expanded })}
            />

            {expanded ? (
              <I18n>Collapse</I18n>
            ) : (
              <div className={clsx('d-none')}>
                <I18n>Expand</I18n>
              </div>
            )}
          </button>
        </li>
      </ul>
    </div>
  );
};

export default Sidebar;
