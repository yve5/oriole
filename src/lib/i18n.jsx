import { I18n, LangSwitch, i18nReducer } from '../shared/i18n';

export { I18n, LangSwitch, i18nReducer };
