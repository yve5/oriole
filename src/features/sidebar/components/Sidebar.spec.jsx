import React from 'react';
import configureMockStore from 'redux-mock-store';

import { thunk } from 'redux-thunk';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { act, fireEvent, render } from '@testing-library/react';

import Sidebar from './Sidebar';
import i18n from '../../../i18n';

const mockStore = configureMockStore([thunk]);

describe('Sidebar', () => {
  it('renders the component', async () => {
    const { asFragment, getByText, rerender } = render(
      <Provider store={mockStore({ i18n })}>
        <MemoryRouter>
          <Sidebar />
        </MemoryRouter>
      </Provider>
    );

    expect(asFragment()).toMatchSnapshot();

    await act(async () => {
      fireEvent.click(getByText('DÃ©velopper'));
    });

    rerender(
      <Provider store={mockStore({ i18n })}>
        <MemoryRouter>
          <Sidebar />
        </MemoryRouter>
      </Provider>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
