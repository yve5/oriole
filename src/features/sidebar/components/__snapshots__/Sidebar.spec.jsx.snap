// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Sidebar > renders the component 1`] = `
<DocumentFragment>
  <div
    class="d-none d-lg-flex flex-column border-end"
  >
    <ul
      class="flex-grow-1 nav nav-pills flex-column mb-auto text-center"
    >
      <li
        class="nav-item"
      >
        <a
          class="nav-link py-3 border-bottom rounded-0 active"
          href="/"
        >
          <svg
            class="svg-icon"
            fill="currentColor"
            height="24"
            viewBox="0 -960 960 960"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M224.62-160q-27.62 0-46.12-18.5Q160-197 160-224.62v-510.76q0-27.62 18.5-46.12Q197-800 224.62-800h510.76q27.62 0 46.12 18.5Q800-763 800-735.38v510.76q0 27.62-18.5 46.12Q763-160 735.38-160H224.62ZM460-383.85H200v159.23q0 10.77 6.92 17.7 6.93 6.92 17.7 6.92H460v-183.85Zm40 0V-200h235.38q10.77 0 17.7-6.92 6.92-6.93 6.92-17.7v-159.23H500Zm-40-40v-185.38H200v185.38h260Zm40 0h260v-185.38H500v185.38ZM200-649.23h560v-86.15q0-10.77-6.92-17.7-6.93-6.92-17.7-6.92H224.62q-10.77 0-17.7 6.92-6.92 6.93-6.92 17.7v86.15Z"
              stroke="currentColor"
              stroke-width="12"
            />
            <title>
              Pagination
            </title>
          </svg>
        </a>
      </li>
      <li
        class="nav-item"
      >
        <a
          class="nav-link py-3 border-bottom rounded-0"
          href="/buttons"
        >
          <svg
            class="svg-icon"
            fill="currentColor"
            height="24"
            viewBox="0 -960 960 960"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M480-320q66.85 0 113.42-46.58Q640-413.15 640-480t-46.58-113.42Q546.85-640 480-640t-113.42 46.58Q320-546.85 320-480t46.58 113.42Q413.15-320 480-320Zm.13 200q-74.67 0-140.41-28.34-65.73-28.34-114.36-76.92-48.63-48.58-76.99-114.26Q120-405.19 120-479.87q0-74.67 28.34-140.41 28.34-65.73 76.92-114.36 48.58-48.63 114.26-76.99Q405.19-840 479.87-840q74.67 0 140.41 28.34 65.73 28.34 114.36 76.92 48.63 48.58 76.99 114.26Q840-554.81 840-480.13q0 74.67-28.34 140.41-28.34 65.73-76.92 114.36-48.58 48.63-114.26 76.99Q554.81-120 480.13-120Zm-.13-40q134 0 227-93t93-227q0-134-93-227t-227-93q-134 0-227 93t-93 227q0 134 93 227t227 93Zm0-320Z"
              stroke="currentColor"
              stroke-width="12"
            />
            <title>
              Boutons
            </title>
          </svg>
        </a>
      </li>
      <li
        class="nav-item"
      >
        <a
          class="nav-link py-3 border-bottom rounded-0"
          href="/buttons"
        >
          <svg
            class="svg-icon"
            fill="currentColor"
            height="24"
            viewBox="0 -960 960 960"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M224.62-160q-27.62 0-46.12-18.5Q160-197 160-224.62v-510.76q0-27.62 18.5-46.12Q197-800 224.62-800h236.15q8.54 0 14.27 5.73t5.73 14.27q0 8.54-5.73 14.27T460.77-760H224.62q-9.24 0-16.93 7.69-7.69 7.69-7.69 16.93v510.76q0 9.24 7.69 16.93 7.69 7.69 16.93 7.69h236.15q8.54 0 14.27 5.73t5.73 14.27q0 8.54-5.73 14.27T460.77-160H224.62Zm497.76-300H387.69q-8.54 0-14.27-5.73T367.69-480q0-8.54 5.73-14.27t14.27-5.73h334.69l-78.84-78.85q-5.62-5.61-6-13.53-.39-7.93 6-14.54 6.38-6.62 14.15-6.73 7.77-.12 14.39 6.5l104.54 104.53q9.69 9.7 9.69 22.62 0 12.92-9.69 22.62L672.08-352.85q-5.85 5.85-13.89 6.12-8.04.27-14.65-6.35-6.39-6.61-6.27-14.27.11-7.65 6.5-14.03L722.38-460Z"
              stroke="currentColor"
              stroke-width="12"
            />
            <title>
              Quelque chose réellement long
            </title>
          </svg>
        </a>
      </li>
    </ul>
    <ul
      class="nav nav-pills flex-column mb-auto text-center"
    >
      <li
        class="nav-item"
      >
        <button
          class="nav-link py-3 rounded-0"
          type="button"
        >
          <svg
            class="svg-icon"
            fill="currentColor"
            height="24"
            viewBox="0 -960 960 960"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M531.69-480 361.85-649.85q-5.62-5.61-6-13.77-.39-8.15 6-14.53 6.38-6.39 14.15-6.39 7.77 0 14.15 6.39l175.54 175.53q5.23 5.24 7.35 10.7 2.11 5.46 2.11 11.92t-2.11 11.92q-2.12 5.46-7.35 10.7L390.15-281.85q-5.61 5.62-13.77 6-8.15.39-14.53-6-6.39-6.38-6.39-14.15 0-7.77 6.39-14.15L531.69-480Z"
              stroke="currentColor"
              stroke-width="12"
            />
          </svg>
          <div
            class="d-none"
          >
            Développer
          </div>
        </button>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`Sidebar > renders the component 2`] = `
<DocumentFragment>
  <div
    class="d-none d-lg-flex flex-column border-end"
  >
    <ul
      class="flex-grow-1 nav nav-pills flex-column mb-auto p-2"
    >
      <li
        class="nav-item"
      >
        <a
          class="nav-link active"
          href="/"
        >
          <svg
            class="svg-icon me-2"
            fill="currentColor"
            height="20"
            viewBox="0 -960 960 960"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M224.62-160q-27.62 0-46.12-18.5Q160-197 160-224.62v-510.76q0-27.62 18.5-46.12Q197-800 224.62-800h510.76q27.62 0 46.12 18.5Q800-763 800-735.38v510.76q0 27.62-18.5 46.12Q763-160 735.38-160H224.62ZM460-383.85H200v159.23q0 10.77 6.92 17.7 6.93 6.92 17.7 6.92H460v-183.85Zm40 0V-200h235.38q10.77 0 17.7-6.92 6.92-6.93 6.92-17.7v-159.23H500Zm-40-40v-185.38H200v185.38h260Zm40 0h260v-185.38H500v185.38ZM200-649.23h560v-86.15q0-10.77-6.92-17.7-6.93-6.92-17.7-6.92H224.62q-10.77 0-17.7 6.92-6.92 6.93-6.92 17.7v86.15Z"
              stroke="currentColor"
              stroke-width="12"
            />
            <title>
              Pagination
            </title>
          </svg>
          Pagination
        </a>
      </li>
      <li
        class="nav-item"
      >
        <a
          class="nav-link"
          href="/buttons"
        >
          <svg
            class="svg-icon me-2"
            fill="currentColor"
            height="20"
            viewBox="0 -960 960 960"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M480-320q66.85 0 113.42-46.58Q640-413.15 640-480t-46.58-113.42Q546.85-640 480-640t-113.42 46.58Q320-546.85 320-480t46.58 113.42Q413.15-320 480-320Zm.13 200q-74.67 0-140.41-28.34-65.73-28.34-114.36-76.92-48.63-48.58-76.99-114.26Q120-405.19 120-479.87q0-74.67 28.34-140.41 28.34-65.73 76.92-114.36 48.58-48.63 114.26-76.99Q405.19-840 479.87-840q74.67 0 140.41 28.34 65.73 28.34 114.36 76.92 48.63 48.58 76.99 114.26Q840-554.81 840-480.13q0 74.67-28.34 140.41-28.34 65.73-76.92 114.36-48.58 48.63-114.26 76.99Q554.81-120 480.13-120Zm-.13-40q134 0 227-93t93-227q0-134-93-227t-227-93q-134 0-227 93t-93 227q0 134 93 227t227 93Zm0-320Z"
              stroke="currentColor"
              stroke-width="12"
            />
            <title>
              Boutons
            </title>
          </svg>
          Boutons
        </a>
      </li>
      <li
        class="nav-item"
      >
        <a
          class="nav-link"
          href="/buttons"
        >
          <svg
            class="svg-icon me-2"
            fill="currentColor"
            height="20"
            viewBox="0 -960 960 960"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M224.62-160q-27.62 0-46.12-18.5Q160-197 160-224.62v-510.76q0-27.62 18.5-46.12Q197-800 224.62-800h236.15q8.54 0 14.27 5.73t5.73 14.27q0 8.54-5.73 14.27T460.77-760H224.62q-9.24 0-16.93 7.69-7.69 7.69-7.69 16.93v510.76q0 9.24 7.69 16.93 7.69 7.69 16.93 7.69h236.15q8.54 0 14.27 5.73t5.73 14.27q0 8.54-5.73 14.27T460.77-160H224.62Zm497.76-300H387.69q-8.54 0-14.27-5.73T367.69-480q0-8.54 5.73-14.27t14.27-5.73h334.69l-78.84-78.85q-5.62-5.61-6-13.53-.39-7.93 6-14.54 6.38-6.62 14.15-6.73 7.77-.12 14.39 6.5l104.54 104.53q9.69 9.7 9.69 22.62 0 12.92-9.69 22.62L672.08-352.85q-5.85 5.85-13.89 6.12-8.04.27-14.65-6.35-6.39-6.61-6.27-14.27.11-7.65 6.5-14.03L722.38-460Z"
              stroke="currentColor"
              stroke-width="12"
            />
            <title>
              Quelque chose réellement long
            </title>
          </svg>
          Quelque chose réellement long
        </a>
      </li>
    </ul>
    <ul
      class="nav nav-pills flex-column mb-auto p-3"
    >
      <li
        class="nav-item"
      >
        <button
          class="nav-link"
          type="button"
        >
          <svg
            class="svg-icon me-2"
            fill="currentColor"
            height="20"
            viewBox="0 -960 960 960"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m404.31-480 169.84 169.85q5.62 5.61 6 13.77.39 8.15-6 14.53-6.38 6.39-14.15 6.39-7.77 0-14.15-6.39L370.31-457.38q-5.23-5.24-7.35-10.7-2.11-5.46-2.11-11.92t2.11-11.92q2.12-5.46 7.35-10.7l175.54-175.53q5.61-5.62 13.77-6 8.15-.39 14.53 6 6.39 6.38 6.39 14.15 0 7.77-6.39 14.15L404.31-480Z"
              stroke="currentColor"
              stroke-width="12"
            />
          </svg>
          Réduire
        </button>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
